pkg/awsconfig/awsconfig.go

```diff
@@ -122,6 +122,15 @@ func (p *CredentialsProvider) Expired() bool {
 	return time.Now().After(creds.Expires)
 }
 
+func (p *CredentialsProvider) ExpiredAt() *time.Time {
```

Comment:
I would recommend passing it as value, since a `time.Time` is a measure of Time.

Reply:
Just noticed :eyes:, can we perhaps use previousCreds.Expires instead of adding ExpiredAt?

Reply:
@mikutas that is indeed a _lot_ simpler :-)